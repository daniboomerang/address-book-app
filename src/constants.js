const SWISS = 'CH';
const SPANISH = 'ES';
const BRITISH = 'BR';
const FRENCH = 'FR';

export { SWISS, SPANISH, BRITISH, FRENCH };
