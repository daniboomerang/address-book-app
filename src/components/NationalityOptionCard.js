import React from 'react';
import cx from 'classnames';
import { SWISS, SPANISH, BRITISH } from '../constants';
import { BrFlag, ChFlag, EsFlag, FrFlag } from './flags';

/**
 * @param {Props} props
 * @param {string} props.nationality - The different nationality options -> CH, ES, BR or FR
 * @param {boolean} props.isSelected - Whether the options shall be deisplayed as selected
 * @param {Function} props.addNationality - Action to add a new nationality to the store
 * @param {Function} props.removeNationality - Action to remove a nationality from the store
 */
const NationalityOptionCard = ({ nationality, isSelected, addNationality, removeNationality }) => {
  const handleAddNationality = () => addNationality(nationality);
  const handleRemoveNationality = () => removeNationality(nationality);

  const renderFlag = () => {
    if (nationality === BRITISH) {
      return <BrFlag className="h-12 w-12 sm:h-24 sm:w-24" />;
    }
    if (nationality === SWISS) {
      return <ChFlag className="h-12 w-12 sm:h-24 sm:w-24" />;
    }
    if (nationality === SPANISH) {
      return <EsFlag className="h-12 w-12 sm:h-24 sm:w-24" />;
    }
    return <FrFlag className="h-12 w-12 sm:h-24 sm:w-24" />;
  };

  return (
    <button
      type="button"
      data-testid={`nationality-${nationality}`}
      onClick={isSelected ? handleRemoveNationality : handleAddNationality}
      className={cx(
        'focus:outline-none grid border-4 text-center justify-center transform cursor-pointer duration-200 hover:border-red-600 hover:text-red-600 pt-4 pb-2 md:pt-5 md:pb-3 w-24 sm:w-32',
        {
          'border-red-600 text-red-600 rounded-md scale-110 hover:border-blue-500 hover:text-blue-500': isSelected,
          'hover:rounded-md border-blue-500': !isSelected,
        }
      )}
    >
      {renderFlag()}
      <span className="mt-4">{nationality}</span>
    </button>
  );
};

export default NationalityOptionCard;
