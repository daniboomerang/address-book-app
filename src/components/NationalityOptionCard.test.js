import React from 'react';
import { mount } from 'enzyme';
import { useDispatch } from 'react-redux';
import NationalityOptionCard from './NationalityOptionCard';
import { SWISS, SPANISH, BRITISH, FRENCH, ADD_NATIONALITY, REMOVE_NATIONALITY } from '../constants';

const mockDispatch = jest.fn();

jest.mock('react-redux', () => ({
  useDispatch: () => mockDispatch,
}));

describe('NationalityOptionCard component', () => {
  const mockedDispatch = jest.fn();
  let wrapper;

  describe('on SWISS nationality option card', () => {
    it('should correctly render and call add nationality action on click', () => {
      wrapper = mount(<NationalityOptionCard nationality={SWISS} isSelected={false} />);

      expect(wrapper.find(`button[data-testid="nationality-${SWISS}"]`).length).toBe(1);
      expect(wrapper.find('.border-blue-500').length).toBe(1);

      wrapper.find(`button[data-testid="nationality-${SWISS}"]`).simulate('click');

      expect(mockDispatch).toHaveBeenCalledWith({ nationality: SWISS, type: ADD_NATIONALITY });
      expect(wrapper).toMatchSnapshot();
    });

    it('should correctly render when selected and call remove nationallity action on click', () => {
      wrapper = mount(<NationalityOptionCard nationality={SWISS} isSelected />);

      expect(wrapper.find(`button[data-testid="nationality-${SWISS}"]`).length).toBe(1);
      expect(wrapper.find('.border-blue-500').length).toBe(0);
      expect(wrapper.find('.text-red-600').length).toBe(1);
      expect(wrapper.find('.border-red-600').length).toBe(1);

      wrapper.find(`button[data-testid="nationality-${SWISS}"]`).simulate('click');

      expect(mockDispatch).toHaveBeenCalledWith({ nationality: SWISS, type: REMOVE_NATIONALITY });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('on BRITISH nationality option card', () => {
    it('should correctly render and call add nationality action on click', () => {
      wrapper = mount(<NationalityOptionCard nationality={BRITISH} isSelected={false} />);

      expect(wrapper.find(`button[data-testid="nationality-${BRITISH}"]`).length).toBe(1);
      expect(wrapper.find('.border-blue-500').length).toBe(1);

      wrapper.find(`button[data-testid="nationality-${BRITISH}"]`).simulate('click');

      expect(mockDispatch).toHaveBeenCalledWith({ nationality: BRITISH, type: ADD_NATIONALITY });
      expect(wrapper).toMatchSnapshot();
    });

    it('should correctly render when selected and call remove nationallity action on click', () => {
      wrapper = mount(<NationalityOptionCard nationality={BRITISH} isSelected />);

      expect(wrapper.find(`button[data-testid="nationality-${BRITISH}"]`).length).toBe(1);
      expect(wrapper.find('.border-blue-500').length).toBe(0);
      expect(wrapper.find('.text-red-600').length).toBe(1);
      expect(wrapper.find('.border-red-600').length).toBe(1);

      wrapper.find(`button[data-testid="nationality-${BRITISH}"]`).simulate('click');

      expect(mockDispatch).toHaveBeenCalledWith({ nationality: BRITISH, type: REMOVE_NATIONALITY });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('on FRENCH nationality option card', () => {
    it('should correctly render and call add nationality action on click', () => {
      wrapper = mount(<NationalityOptionCard nationality={FRENCH} isSelected={false} />);

      expect(wrapper.find(`button[data-testid="nationality-${FRENCH}"]`).length).toBe(1);
      expect(wrapper.find('.border-blue-500').length).toBe(1);

      wrapper.find(`button[data-testid="nationality-${FRENCH}"]`).simulate('click');
      expect(mockDispatch).toHaveBeenCalledWith({ nationality: FRENCH, type: ADD_NATIONALITY });
      expect(wrapper).toMatchSnapshot();
    });

    it('should correctly render when selected and call remove nationallity action on click', () => {
      wrapper = mount(<NationalityOptionCard nationality={FRENCH} isSelected />);

      expect(wrapper.find(`button[data-testid="nationality-${FRENCH}"]`).length).toBe(1);
      expect(wrapper.find('.border-blue-500').length).toBe(0);
      expect(wrapper.find('.text-red-600').length).toBe(1);
      expect(wrapper.find('.border-red-600').length).toBe(1);

      wrapper.find(`button[data-testid="nationality-${FRENCH}"]`).simulate('click');

      expect(mockDispatch).toHaveBeenCalledWith({ nationality: FRENCH, type: ADD_NATIONALITY });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('on SPANISH nationality option card', () => {
    it('should correctly render and call add nationality action on click', () => {
      wrapper = mount(<NationalityOptionCard nationality={SPANISH} isSelected={false} />);

      expect(wrapper.find(`button[data-testid="nationality-${SPANISH}"]`).length).toBe(1);
      expect(wrapper.find('.border-blue-500').length).toBe(1);

      wrapper.find(`button[data-testid="nationality-${SPANISH}"]`).simulate('click');

      expect(mockDispatch).toHaveBeenCalledWith({ nationality: SPANISH, type: ADD_NATIONALITY });
      expect(wrapper).toMatchSnapshot();
    });

    it('should correctly render when selected and call remove nationallity action on click', () => {
      wrapper = mount(<NationalityOptionCard nationality={SPANISH} isSelected />);

      expect(wrapper.find(`button[data-testid="nationality-${SPANISH}"]`).length).toBe(1);
      expect(wrapper.find('.border-blue-500').length).toBe(0);
      expect(wrapper.find('.text-red-600').length).toBe(1);
      expect(wrapper.find('.border-red-600').length).toBe(1);

      wrapper.find(`button[data-testid="nationality-${SPANISH}"]`).simulate('click');

      expect(mockDispatch).toHaveBeenCalledWith({ nationality: SPANISH, type: ADD_NATIONALITY });
      expect(wrapper).toMatchSnapshot();
    });
  });
});
