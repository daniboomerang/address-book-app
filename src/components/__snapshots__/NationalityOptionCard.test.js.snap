// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NationalityOptionCard component on BRITISH nationality option card should correctly render and call add nationality action on click 1`] = `
<NationalityOptionCard
  addNationality={
    [MockFunction] {
      "calls": Array [
        Array [
          "CH",
        ],
        Array [
          "BR",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  isSelected={false}
  nationality="BR"
  removeNationality={
    [MockFunction] {
      "calls": Array [
        Array [
          "CH",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <button
    className="focus:outline-none grid border-4 text-center justify-center transform cursor-pointer duration-200 hover:border-red-600 hover:text-red-600 pt-4 pb-2 md:pt-5 md:pb-3 w-24 sm:w-32 hover:rounded-md border-blue-500"
    data-testid="nationality-BR"
    onClick={[Function]}
    type="button"
  >
    <BrFlag
      className="h-12 w-12 sm:h-24 sm:w-24"
    >
      <svg
        className="h-12 w-12 sm:h-24 sm:w-24"
        data-testid="BR-flag"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="#F0F0F0"
          height="512"
          width="512"
        />
        <polygon
          fill="#D80027"
          points="0,304 208,304 208,512 304,512 304,304 512,304 512,208 304,208 304,0 208,0 208,208 0,208   "
        />
        <g>
          <polygon
            fill="#0052B4"
            points="406.92,333.913 512,438.993 512,333.913  "
          />
          <polygon
            fill="#0052B4"
            points="333.913,333.913 512,512 512,461.64 384.273,333.913  "
          />
          <polygon
            fill="#0052B4"
            points="464.564,512 333.913,381.336 333.913,512  "
          />
        </g>
        <polygon
          fill="#F0F0F0"
          points="333.913,333.913 512,512 512,461.64 384.273,333.913 "
        />
        <polygon
          fill="#D80027"
          points="333.913,333.913 512,512 512,461.64 384.273,333.913 "
        />
        <g>
          <polygon
            fill="#0052B4"
            points="80.302,333.913 0,414.215 0,333.913  "
          />
          <polygon
            fill="#0052B4"
            points="178.084,356.559 178.084,511.997 22.658,511.997  "
          />
        </g>
        <polygon
          fill="#D80027"
          points="127.724,333.916 0,461.641 0,512 0,512 178.084,333.916 "
        />
        <g>
          <polygon
            fill="#0052B4"
            points="105.08,178.087 0,73.007 0,178.087  "
          />
          <polygon
            fill="#0052B4"
            points="178.087,178.087 0,0 0,50.36 127.727,178.087  "
          />
          <polygon
            fill="#0052B4"
            points="47.436,0 178.087,130.663 178.087,0  "
          />
        </g>
        <polygon
          fill="#F0F0F0"
          points="178.087,178.087 0,0 0,50.36 127.727,178.087 "
        />
        <polygon
          fill="#D80027"
          points="178.087,178.087 0,0 0,50.36 127.727,178.087 "
        />
        <g>
          <polygon
            fill="#0052B4"
            points="431.698,178.087 512,97.785 512,178.087  "
          />
          <polygon
            fill="#0052B4"
            points="333.916,155.441 333.916,0.003 489.342,0.003  "
          />
        </g>
        <polygon
          fill="#D80027"
          points="384.276,178.084 512,50.359 512,0 512,0 333.916,178.084 "
        />
      </svg>
    </BrFlag>
    <span
      className="mt-4"
    >
      BR
    </span>
  </button>
</NationalityOptionCard>
`;

exports[`NationalityOptionCard component on BRITISH nationality option card should correctly render when selected and call remove nationallity action on click 1`] = `
<NationalityOptionCard
  addNationality={
    [MockFunction] {
      "calls": Array [
        Array [
          "CH",
        ],
        Array [
          "BR",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  isSelected={true}
  nationality="BR"
  removeNationality={
    [MockFunction] {
      "calls": Array [
        Array [
          "CH",
        ],
        Array [
          "BR",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <button
    className="focus:outline-none grid border-4 text-center justify-center transform cursor-pointer duration-200 hover:border-red-600 hover:text-red-600 pt-4 pb-2 md:pt-5 md:pb-3 w-24 sm:w-32 border-red-600 text-red-600 rounded-md scale-110 hover:border-blue-500 hover:text-blue-500"
    data-testid="nationality-BR"
    onClick={[Function]}
    type="button"
  >
    <BrFlag
      className="h-12 w-12 sm:h-24 sm:w-24"
    >
      <svg
        className="h-12 w-12 sm:h-24 sm:w-24"
        data-testid="BR-flag"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="#F0F0F0"
          height="512"
          width="512"
        />
        <polygon
          fill="#D80027"
          points="0,304 208,304 208,512 304,512 304,304 512,304 512,208 304,208 304,0 208,0 208,208 0,208   "
        />
        <g>
          <polygon
            fill="#0052B4"
            points="406.92,333.913 512,438.993 512,333.913  "
          />
          <polygon
            fill="#0052B4"
            points="333.913,333.913 512,512 512,461.64 384.273,333.913  "
          />
          <polygon
            fill="#0052B4"
            points="464.564,512 333.913,381.336 333.913,512  "
          />
        </g>
        <polygon
          fill="#F0F0F0"
          points="333.913,333.913 512,512 512,461.64 384.273,333.913 "
        />
        <polygon
          fill="#D80027"
          points="333.913,333.913 512,512 512,461.64 384.273,333.913 "
        />
        <g>
          <polygon
            fill="#0052B4"
            points="80.302,333.913 0,414.215 0,333.913  "
          />
          <polygon
            fill="#0052B4"
            points="178.084,356.559 178.084,511.997 22.658,511.997  "
          />
        </g>
        <polygon
          fill="#D80027"
          points="127.724,333.916 0,461.641 0,512 0,512 178.084,333.916 "
        />
        <g>
          <polygon
            fill="#0052B4"
            points="105.08,178.087 0,73.007 0,178.087  "
          />
          <polygon
            fill="#0052B4"
            points="178.087,178.087 0,0 0,50.36 127.727,178.087  "
          />
          <polygon
            fill="#0052B4"
            points="47.436,0 178.087,130.663 178.087,0  "
          />
        </g>
        <polygon
          fill="#F0F0F0"
          points="178.087,178.087 0,0 0,50.36 127.727,178.087 "
        />
        <polygon
          fill="#D80027"
          points="178.087,178.087 0,0 0,50.36 127.727,178.087 "
        />
        <g>
          <polygon
            fill="#0052B4"
            points="431.698,178.087 512,97.785 512,178.087  "
          />
          <polygon
            fill="#0052B4"
            points="333.916,155.441 333.916,0.003 489.342,0.003  "
          />
        </g>
        <polygon
          fill="#D80027"
          points="384.276,178.084 512,50.359 512,0 512,0 333.916,178.084 "
        />
      </svg>
    </BrFlag>
    <span
      className="mt-4"
    >
      BR
    </span>
  </button>
</NationalityOptionCard>
`;

exports[`NationalityOptionCard component on FRENCH nationality option card should correctly render and call add nationality action on click 1`] = `
<NationalityOptionCard
  addNationality={
    [MockFunction] {
      "calls": Array [
        Array [
          "CH",
        ],
        Array [
          "BR",
        ],
        Array [
          "FR",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  isSelected={false}
  nationality="FR"
  removeNationality={
    [MockFunction] {
      "calls": Array [
        Array [
          "CH",
        ],
        Array [
          "BR",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <button
    className="focus:outline-none grid border-4 text-center justify-center transform cursor-pointer duration-200 hover:border-red-600 hover:text-red-600 pt-4 pb-2 md:pt-5 md:pb-3 w-24 sm:w-32 hover:rounded-md border-blue-500"
    data-testid="nationality-FR"
    onClick={[Function]}
    type="button"
  >
    <FrFlag
      className="h-12 w-12 sm:h-24 sm:w-24"
    >
      <svg
        className="h-12 w-12 sm:h-24 sm:w-24"
        data-testid="FR-flag"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="#F0F0F0"
          height="512"
          width="512"
        />
        <rect
          fill="#0052B4"
          height="512"
          width="170.663"
        />
        <rect
          fill="#D80027"
          height="512"
          width="170.663"
          x="341.337"
        />
      </svg>
    </FrFlag>
    <span
      className="mt-4"
    >
      FR
    </span>
  </button>
</NationalityOptionCard>
`;

exports[`NationalityOptionCard component on FRENCH nationality option card should correctly render when selected and call remove nationallity action on click 1`] = `
<NationalityOptionCard
  addNationality={
    [MockFunction] {
      "calls": Array [
        Array [
          "CH",
        ],
        Array [
          "BR",
        ],
        Array [
          "FR",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  isSelected={true}
  nationality="FR"
  removeNationality={
    [MockFunction] {
      "calls": Array [
        Array [
          "CH",
        ],
        Array [
          "BR",
        ],
        Array [
          "FR",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <button
    className="focus:outline-none grid border-4 text-center justify-center transform cursor-pointer duration-200 hover:border-red-600 hover:text-red-600 pt-4 pb-2 md:pt-5 md:pb-3 w-24 sm:w-32 border-red-600 text-red-600 rounded-md scale-110 hover:border-blue-500 hover:text-blue-500"
    data-testid="nationality-FR"
    onClick={[Function]}
    type="button"
  >
    <FrFlag
      className="h-12 w-12 sm:h-24 sm:w-24"
    >
      <svg
        className="h-12 w-12 sm:h-24 sm:w-24"
        data-testid="FR-flag"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="#F0F0F0"
          height="512"
          width="512"
        />
        <rect
          fill="#0052B4"
          height="512"
          width="170.663"
        />
        <rect
          fill="#D80027"
          height="512"
          width="170.663"
          x="341.337"
        />
      </svg>
    </FrFlag>
    <span
      className="mt-4"
    >
      FR
    </span>
  </button>
</NationalityOptionCard>
`;

exports[`NationalityOptionCard component on SPANISH nationality option card should correctly render and call add nationality action on click 1`] = `
<NationalityOptionCard
  addNationality={
    [MockFunction] {
      "calls": Array [
        Array [
          "CH",
        ],
        Array [
          "BR",
        ],
        Array [
          "FR",
        ],
        Array [
          "ES",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  isSelected={false}
  nationality="ES"
  removeNationality={
    [MockFunction] {
      "calls": Array [
        Array [
          "CH",
        ],
        Array [
          "BR",
        ],
        Array [
          "FR",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <button
    className="focus:outline-none grid border-4 text-center justify-center transform cursor-pointer duration-200 hover:border-red-600 hover:text-red-600 pt-4 pb-2 md:pt-5 md:pb-3 w-24 sm:w-32 hover:rounded-md border-blue-500"
    data-testid="nationality-ES"
    onClick={[Function]}
    type="button"
  >
    <BrFlag
      className="h-12 w-12 sm:h-24 sm:w-24"
    >
      <svg
        className="h-12 w-12 sm:h-24 sm:w-24"
        data-testid="ES-flag"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="#FFDA44"
          height="512"
          width="512"
        />
        <g>
          <rect
            fill="#D80027"
            height="170.663"
            width="512"
          />
          <rect
            fill="#D80027"
            height="170.663"
            width="512"
            y="341.337"
          />
        </g>
      </svg>
    </BrFlag>
    <span
      className="mt-4"
    >
      ES
    </span>
  </button>
</NationalityOptionCard>
`;

exports[`NationalityOptionCard component on SPANISH nationality option card should correctly render when selected and call remove nationallity action on click 1`] = `
<NationalityOptionCard
  addNationality={
    [MockFunction] {
      "calls": Array [
        Array [
          "CH",
        ],
        Array [
          "BR",
        ],
        Array [
          "FR",
        ],
        Array [
          "ES",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  isSelected={true}
  nationality="ES"
  removeNationality={
    [MockFunction] {
      "calls": Array [
        Array [
          "CH",
        ],
        Array [
          "BR",
        ],
        Array [
          "FR",
        ],
        Array [
          "ES",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <button
    className="focus:outline-none grid border-4 text-center justify-center transform cursor-pointer duration-200 hover:border-red-600 hover:text-red-600 pt-4 pb-2 md:pt-5 md:pb-3 w-24 sm:w-32 border-red-600 text-red-600 rounded-md scale-110 hover:border-blue-500 hover:text-blue-500"
    data-testid="nationality-ES"
    onClick={[Function]}
    type="button"
  >
    <BrFlag
      className="h-12 w-12 sm:h-24 sm:w-24"
    >
      <svg
        className="h-12 w-12 sm:h-24 sm:w-24"
        data-testid="ES-flag"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="#FFDA44"
          height="512"
          width="512"
        />
        <g>
          <rect
            fill="#D80027"
            height="170.663"
            width="512"
          />
          <rect
            fill="#D80027"
            height="170.663"
            width="512"
            y="341.337"
          />
        </g>
      </svg>
    </BrFlag>
    <span
      className="mt-4"
    >
      ES
    </span>
  </button>
</NationalityOptionCard>
`;

exports[`NationalityOptionCard component on SWISS nationality option card should correctly render and call add nationality action on click 1`] = `
<NationalityOptionCard
  addNationality={
    [MockFunction] {
      "calls": Array [
        Array [
          "CH",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  isSelected={false}
  nationality="CH"
  removeNationality={[MockFunction]}
>
  <button
    className="focus:outline-none grid border-4 text-center justify-center transform cursor-pointer duration-200 hover:border-red-600 hover:text-red-600 pt-4 pb-2 md:pt-5 md:pb-3 w-24 sm:w-32 hover:rounded-md border-blue-500"
    data-testid="nationality-CH"
    onClick={[Function]}
    type="button"
  >
    <BrFlag
      className="h-12 w-12 sm:h-24 sm:w-24"
    >
      <svg
        className="h-12 w-12 sm:h-24 sm:w-24"
        data-testid="CH-flag"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="#D80027"
          height="512"
          width="512"
        />
        <polygon
          fill="#F0F0F0"
          points="395.13,209.624 302.377,209.624 302.377,116.87 209.623,116.87 209.623,209.624   116.87,209.624 116.87,302.377 209.623,302.377 209.623,395.13 302.377,395.13 302.377,302.377 395.13,302.377 "
        />
      </svg>
    </BrFlag>
    <span
      className="mt-4"
    >
      CH
    </span>
  </button>
</NationalityOptionCard>
`;

exports[`NationalityOptionCard component on SWISS nationality option card should correctly render when selected and call remove nationallity action on click 1`] = `
<NationalityOptionCard
  addNationality={
    [MockFunction] {
      "calls": Array [
        Array [
          "CH",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  isSelected={true}
  nationality="CH"
  removeNationality={
    [MockFunction] {
      "calls": Array [
        Array [
          "CH",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <button
    className="focus:outline-none grid border-4 text-center justify-center transform cursor-pointer duration-200 hover:border-red-600 hover:text-red-600 pt-4 pb-2 md:pt-5 md:pb-3 w-24 sm:w-32 border-red-600 text-red-600 rounded-md scale-110 hover:border-blue-500 hover:text-blue-500"
    data-testid="nationality-CH"
    onClick={[Function]}
    type="button"
  >
    <BrFlag
      className="h-12 w-12 sm:h-24 sm:w-24"
    >
      <svg
        className="h-12 w-12 sm:h-24 sm:w-24"
        data-testid="CH-flag"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="#D80027"
          height="512"
          width="512"
        />
        <polygon
          fill="#F0F0F0"
          points="395.13,209.624 302.377,209.624 302.377,116.87 209.623,116.87 209.623,209.624   116.87,209.624 116.87,302.377 209.623,302.377 209.623,395.13 302.377,395.13 302.377,302.377 395.13,302.377 "
        />
      </svg>
    </BrFlag>
    <span
      className="mt-4"
    >
      CH
    </span>
  </button>
</NationalityOptionCard>
`;
